src/allmydata/scripts/tahoe_ls.py

```diff
@@ -185,12 +189,19 @@ def list(options):
     fmt = " ".join(fmt_pieces)
 
     rc = 0
-    for (encoding_error, row) in rows:
+    for row in rows:
+        row = (fmt % tuple(row)).rstrip()
+        encoding_error = False
+        try:
+            row = unicode_to_output(row)
+        except UnicodeEncodeError:
+            encoding_error = True
+            row = quote_output(row)
         if encoding_error:
-            print((fmt % tuple(row)).rstrip(), file=stderr)
+            print(row, file=stderr)
```

Comment:
It doesn't look like this code path ever had unit test coverage.

https://coveralls.io/jobs/81285335/source_files/9127104468#L197

I wonder what such a test would look like, and if writing that would be a worthwhile endeavor.  But that probably would be outside the scope of this PR.  From 496802420ebd25da630d516880e3f4c6c8258e97, it seems this path is triggered from an unhappy mix of incompatible system locale and filesystem encoding, and that is hard to test.

Perhaps this code path too would be irrelevant once Python 2 is dropped?  `"Ärtonwall".encode("utf-8")` throws `UnicodeDecodeError` only under Python 2, for example.